### ISSUE TITLE
document the array-of-structure and structure-of-array layout

### ISSUE THREAD

**Issue by johnnychen94** [JuliaImages/juliaimages.github.io#154]
JuliaImages uses the array of structure layout to fully use the multiple dispatch mechanism without introducing too many ambiguities. Intrinsically, this array-of-structure layout is slower than structure-of-array layout when we need a contiguous memory per channel.

For example, compute the norm of a RGB image:

```julia
julia> x = rand(RGB{Float64}, 100, 100);

julia> cx = collect(channelview(x));

julia> @btime norm($x)
  28.822 μs (0 allocations: 0 bytes)
100.60011111008363

julia> @btime norm($cx)
  7.552 μs (0 allocations: 0 bytes)
100.60011111008343
```

I was only aware of this issue months ago and still not very familiar with this topic, @timholy would you mind giving a section or page documenting aspects that we need to be aware of wrt this? These differences might be emphasized in the docs because most other image processing libraries use the structure-of-array layout.

**Comment by timholy** [comment]
Yes, we should document this. But I am actually gradually working on this area now (the `ReinterpretArray` work is part of it) and I am planning to provide better support for other layouts.
